lib/bolt/applicator.rb

```diff
@@ -272,8 +277,8 @@ def apply_ast(raw_ast, targets, options, plan_vars = {})
       r
     end
 
-    def plugins
-      @plugin_tarball.value ||
+    def plugins(for_target_node)
+      for_target_node.value ||
```

Comment:
```suggestion
    def plugins(plugin_tarball)
      plugin_tarball.value ||
```

`for_target_node` seems ambiguous / misleading on what the parameter actually is.